app/src/main/java/com/futurice/freesound/viewmodel/DisplayableItem.java

```diff
@@ -0,0 +1,46 @@
+package com.futurice.freesound.viewmodel;
+
+import com.google.auto.value.AutoValue;
+
+import android.support.annotation.NonNull;
+
+/**
+ * Wraps a model. Convenient for complex presentation layers such as
+ * {@link android.support.v7.widget.RecyclerView} where different types of model are handled.
+ */
+@AutoValue
+public abstract class DisplayableItem<T> {
+
+    // Types
+    public static final int SOUND = 0;
```

Comment:
I tried experimenting with the [Android support annotations](https://developer.android.com/studio/write/annotations.html) to see if they could be used to enforce the kind of type safety that I think using plain old ints lacks. I wrote the code block below to test if defining two constants with the same values inside an `@IntDef` would cause a compiler error - sadly it doesn't.

```java
    @IntDef({SOUND, AD})
    @Retention(RetentionPolicy.SOURCE)
    public @interface ItemType {}

    public static final int SOUND = 0;
    public static final int AD = 0;

    @ItemType
    public abstract int type();
```
This code compiles, which is not great for type safety.

In this case, the choice of enum or int is a choice between type safety and simplicity.